File: python/Neetcode/LinkedList/Easy/203.RemoveLinkedListElements.py
# Given the head of a linked list and an integer val, 
# remove all the nodes of the linked list that has Node.val == val, 
# and return the new head.

class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def removeElements(head, val):
    new_list = ListNode(next=head)
    previous = new_list
    current = head

    while current:
        if current.val == val:
            previous.next = current.next
        else:
            previous = current
        
        current = current.next

    return new_list.next

File: swift/Neetcode/LinkedList/Easy/203.RemoveLinkedListElements.swift
// Given the head of a linked list and an integer val, 
// remove all the nodes of the linked list that has Node.val == val, 
// and return the new head.

func removeElements(_ head: ListNode?, _ val: Int) -> ListNode? {
    let newHead = ListNode(0, head)
    var previous: ListNode? = newHead
    var current = head

    while let node = current {
        let next = node.next

        if node.val == val {
            previous?.next = next
        } else {
            previous = current
        }

        current = next
    }

    return newHead.next
}